tests/cpp/test_combined_inner_outer_reduction.cpp

```diff
@@ -34,255 +34,129 @@ namespace nvfuser {
+    layer_norm_results.grad_weight =
+        castOp(dtype, layer_norm_results.grad_weight);
+  }
 
-    testValidate(
-        &fusion,
-        {cg_outputs[0], cg_outputs[1], cg_outputs[2]},
-        aten_inputs,
-        {std::get<0>(aten_gradients),
-         std::get<1>(aten_gradients),
-         std::get<2>(aten_gradients)},
-        __LINE__,
-        __FILE__);
-    if (isBenchmark) {
-      FusionKernelRuntime* fkr = fec.getMostRecentKernelRuntime();
-      fkr->enableKernelTimeMeasurement();
-
-      constexpr int nwarm = 5;
-      constexpr int niter = 10;
-      std::vector<float> bw(niter, 0.f);
-      std::vector<float> timeus(niter, 0.f);
-
-      size_t read_write_bytes = 0;
-      const std::vector<at::Tensor> aten_inputs_tmp = {
-          aten_grad_out,
-          aten_input,
-          aten_mean,
-          aten_rstd,
-          aten_weight,
-          aten_bias};
-      const std::vector<at::Tensor> aten_output_tmp = {
-          std::get<0>(aten_gradients),
-          std::get<1>(aten_gradients),
-          std::get<2>(aten_gradients)};
-      for (auto input : aten_inputs_tmp) {
-        read_write_bytes += input.numel() * input.element_size();
-      }
-      for (auto output : aten_output_tmp) {
-        read_write_bytes += output.numel() * output.element_size();
-      }
-
-      for (int i = 0; i < nwarm + niter; i++) {
-        clearL2Cache();
-        // fe.runFusion(inputs, outputs, launch_constraints);
-        auto cg_outputs = fec.runFusionWithInputs(aten_inputs);
-        if (i >= nwarm) {
-          float runTimeus = 0.0f;
-          int num_kernels = fkr->executors().size();
-          for (int i = 0; i < num_kernels; i++) {
-            const FusionExecutor& fe = fkr->executors()[i];
-            runTimeus += fe.kernelTimeMs() * 1e3;
-          }
-          float bandwidth = read_write_bytes / 1e9 / (runTimeus * 1e-6);
-          timeus[i - nwarm] = runTimeus;
-          bw[i - nwarm] = bandwidth;
-          if (verbose == 2)
-            std::cout << "iter= " << i << ", bandwidth= " << bandwidth << "GB/s"
-                      << ", time= " << runTimeus << " us" << std::endl;
-        }
-      }
-      return getMeanVar(timeus);
-    } else {
-      if (verbose == 1) {
-        std::stringstream sdim0, sdim1;
-        std::for_each(
-            batch_shape.begin(), batch_shape.end(), [&sdim0](int64_t n) {
-              sdim0 << n << " x ";
-            });
-        std::for_each(
-            norm_shape.begin(), norm_shape.end(), [&sdim1](int64_t n) {
-              sdim1 << n << " x ";
-            });
-        std::string str1 = sdim1.str();
-        str1.erase(str1.end() - 2);
-        std::cout << "passed, shape= " << sdim0.str() << str1 << std::endl;
-      }
-      return std::make_tuple(-1.0f, -1.0f);
-    }
-  };
+  fusion.addOutput(layer_norm_results.grad_input);
+  fusion.addOutput(layer_norm_results.grad_weight);
+  fusion.addOutput(layer_norm_results.grad_bias);
+
+  auto maybe_fp16_options =
```

Comment:
```suggestion
  auto options =
```
I don't see the value of mentioning `maybe_fp16`. Obviously, `dtype` can be fp16 and `options` just follow whatever dtype that's passed in.